llama_index/vector_stores/utils.py

```diff
@@ -0,0 +1,62 @@
+
+
+import json
+from dataclasses import field
+from typing import Dict, Tuple
+
+from llama_index.data_structs.node import DocumentRelationship, Node
+
+
+def node_to_metadata_dict(node: Node) -> dict:
+    """Common logic for saving Node data into metadata dict."""
+    metadata = {}
+    
+    # store extra_info directly to allow metadata filtering
+    metadata.update(node.extra_info)
+
+    # json-serialize the node_info
+    node_info_str = ""
+    if node.node_info is not None:
+        node_info_str = json.dumps(node.node_info)
+    metadata["node_info"] = node_info_str
+
+    # json-serialize the relationships
+    relationships_str = ""
+    if node.relationships is not None:
+        relationships_str = json.dumps(node.relationships)
+    metadata["relationships"] = relationships_str
+
+    # store ref doc id at top level to allow metadata filtering
+    metadata['document_id'] = node.ref_doc_id or "None" # for Chroma
+    metadata['doc_id'] = node.ref_doc_id or "None"  # for Pinecone, Qdrant
+    metadata['ref_doc_id'] = node.ref_doc_id or "None" # for Weaviate
```

Comment:
No way to align the id here for all indexes hey?

Reply:
Only left for backward compatibility. The way we did things was a bit all over the place.